components/fxa-client/src/http_client.rs

```diff
@@ -104,7 +106,17 @@ pub trait FxAClient {
     ) -> Result<HashMap<String, ScopedKeyDataResponse>>;
 }
 
-pub struct Client;
+enum FxAClientState {
+    Ok,
+    TimeOut {
+        time_out_end_duration: Duration,
```

Comment:
It could be that different endpoints we have will have different `retryAfter` limits. Is it possible to HashMap with a { 
endpoint: time_out_end_time
}
Example draft: `/commands` : `1591892366`